kotlin/workflow-ui-android/src/main/java/com/squareup/workflow/ui/ModalContainer.kt

```diff
@@ -261,20 +261,20 @@ private class OnDestroy(private val block: () -> Unit) : LifecycleObserver {
   fun onDestroy() = block()
 }
 
-private val Dialog.activity: AppCompatActivity?
-  get() = decorView?.context?.activity()
+private val Dialog.lifecycle: Lifecycle?
```

Comment:
Let's be consistent about whether `lifecycle` is a property or a function. I could go either way – it returns consistently for a given receiver, but also involves a potentially expensive search. But either way, let's be consistent.

Reply:
Fair enough. Looks like `tailrec` isn't available for property getters, so I'll make them all functions.

Reply:
Ah, I forgot about that. That's a good enough reason to make them inconsistent, just document it.

Reply:
Renamed to `lifecycleOrNull()`.